app-text/aspell/files/aspell-0.60.6.1-unicode.patch

```diff
@@ -0,0 +1,299 @@
+-       fi
+-
+-     fi
+- 
+-     AC_MSG_CHECKING(if standard curses include sequence will work)
+-     AC_TRY_LINK(
+-       [#ifdef DEFINE_XOPEN_SOURCE_EXTENDED
+-        #  define _XOPEN_SOURCE_EXTENDED 1
+-        #endif
+-        #include <termios.h>
+-	#include <unistd.h>
+-        #include CURSES_HEADER
+-	#include TERM_HEADER
+-       ],
+-       [tigetstr(const_cast<char *>("cup"));],
+-       [AC_MSG_RESULT(yes)
+-	AC_DEFINE(HAVE_LIBCURSES, 1, 
+-                  [Defined if the curses library is available])
+-	posix_termios=t
+-	AC_DEFINE(CURSES_INCLUDE_STANDARD, 1, 
+-                  [Defined if no special Workarounds are needed for Curses headers])],
+-       [AC_MSG_RESULT(no)
+-
+-     dnl else if
+-     AC_MSG_CHECKING(if curses workaround I will work)
+-     AC_TRY_LINK(
+-       [#ifdef DEFINE_XOPEN_SOURCE_EXTENDED
+-        #  define _XOPEN_SOURCE_EXTENDED 1
+-        #endif
+-        #include <termios.h>
+-	#include <unistd.h>
+-        #include CURSES_HEADER
+-	extern "C" {char * tigetstr(char * capname);}],
+-       [tigetstr(const_cast<char *>("cup"));],
+-       [AC_MSG_RESULT(yes)
+-	AC_DEFINE(HAVE_LIBCURSES, 1, [])
+-        posix_termios=t
+-	AC_DEFINE(CURSES_INCLUDE_WORKAROUND_1, 1, 
+-                  [Defined if special Wordaround I is need for Curses headers])],
+-       [AC_MSG_RESULT(no)
+-
+-     dnl else if
+-     AC_MSG_CHECKING(if curses without Unix stuff will work)
+-     AC_TRY_LINK(
+-       [#include CURSES_HEADER
+-       ],
+-       [initscr();],
+-       [AC_MSG_RESULT(yes)
+-	AC_DEFINE(HAVE_LIBCURSES, 1, [])
+-	AC_DEFINE(CURSES_ONLY, 1, 
+-                  [Defined if curses like POSIX Functions should be used])
+-        curses_only=t],
+-       [AC_MSG_RESULT(no)
+-
+-     dnl else
+-     use_curses=false
+-     CURSES_LIBS=""
+-     CURSES_INCLUDE=""
+-
+-     ]) ]) ])
++  dnl else if
++  AC_MSG_CHECKING(if curses workaround I will work)
++  AC_TRY_LINK(
++    [#ifdef DEFINE_XOPEN_SOURCE_EXTENDED
++     #  define _XOPEN_SOURCE_EXTENDED 1
++     #endif
++     #include <termios.h>
++     #include <unistd.h>
++     #include CURSES_HEADER
++     extern "C" {char * tigetstr(char * capname);}],
++    [tigetstr(const_cast<char *>("cup"));],
++    [AC_MSG_RESULT(yes)
++     AC_DEFINE(HAVE_LIBCURSES, 1, [])
++     posix_termios=t
++     AC_DEFINE(CURSES_INCLUDE_WORKAROUND_1, 1, 
++               [Defined if special Wordaround I is need for Curses headers])],
++    [AC_MSG_RESULT(no)
+ 
+-  fi
++  dnl else if
++  AC_MSG_CHECKING(if curses without Unix stuff will work)
++  AC_TRY_LINK(
++    [#include CURSES_HEADER
++    ],
++    [initscr();],
++    [AC_MSG_RESULT(yes)
++     AC_DEFINE(HAVE_LIBCURSES, 1, [])
++     AC_DEFINE(CURSES_ONLY, 1, 
++               [Defined if curses like POSIX Functions should be used])
++     curses_only=t],
++    [AC_MSG_RESULT(no)
++
++  dnl else
++  use_curses=false
++  CURSES_LIBS=""
++  CURSES_INCLUDE=""
++
++  ]) ]) ])
```

Comment:
The problem is that the parts you have misgivings about define Autoconf variables (like `CURSES_ONLY`) which influence which parts of the source code are compiled. I tried to rip out as much as possible from `configure.ac` without influencing `confdefs.h`. Another issue is that it looks like my patch adds a lot of crap to `configure.ac`, but those are in fact whitespace changes. You can see all the changes made by this patch excluding whitespace changes [here](http://pastebin.com/FYR6w9uQ).

I can do one of the following:

- edit the patch so that it does not do whitespace changes,
- nuke `configure.ac` from orbit, ripping out even more stuff, but possibly changing the contents of `confdefs.h` and thus the resulting binaries.

Which one would you prefer? Or perhaps you see other options?